python/ml4ir/base/features/feature_config.py

```diff
@@ -938,6 +978,10 @@ def get_shape(feature_info: dict):
             # Setting size to None for sequence features as the num_records is variable
             if feature_info["tfrecord_type"] == SequenceExampleTypeKey.SEQUENCE:
                 return feature_info.get("shape", (None,))
+            # The fr feature size is pre-defined to max_ranks
+            elif feature_info["node_name"] == "fr_feature":
```

Comment:
@mohazahran Remove `fr` here as well. You should call it `rank`

Reply:
You are also doing a check on a `node_name`. I don't think this is good coding practice. We are trying to keep `ml4ir` as generic as possible and the only hard-coded keys we expect are `query_key`, `rank`, `label` and `features`.

Reply:
I'm checking the node's name because the 'fr' column is being used now as a rank and as a feature for positional bias.